pkg/cmd/pulumi/deployment_settings_config.go

```diff
@@ -971,19 +934,75 @@ func configureImageRepository(ctx context.Context, displayOpts *display.Options,
 	return nil
 }
 
-func askForConfirmation(prompt string, color colors.Colorization, defaultValue bool) (bool, error) {
-	var option string
+func askForConfirmation(prompt string, color colors.Colorization, defaultValue bool, yes bool) bool {
 	def := optNo
 	if defaultValue {
 		def = optYes
 	}
-	if err := survey.AskOne(&survey.Select{
-		Message: prompt,
-		Options: []string{optYes, optNo},
-		Default: def,
-	}, &option, surveyIcons(color)); err != nil {
-		return false, errors.New("selection cancelled")
+	options := []string{optYes, optNo}
+	response := promptUserSkippable(yes, prompt, options, def, color)
+	return response == optYes
+}
+
+// ValidateRelativeDirectory ensures a relative path points to a valid directory
+func ValidateRelativeDirectory(rootDir string) func(string) error {
+	return func(s string) error {
+		if rootDir == "" {
+			return nil
+		}
+
+		dir := path.Join(rootDir, s)
+		info, err := os.Stat(dir)
+
+		switch {
+		case os.IsNotExist(err):
+			return fmt.Errorf("invalid relative path %s", s)
+		case err != nil:
+			return err
+		}
+
+		if !info.IsDir() {
+			return fmt.Errorf("invalid relative path %s, is not a directory", s)
+		}
+
+		return nil
+	}
+}
+
+func ValidateBranchName(s string) error {
+	const maxTagValueLength = 256
+
+	if s == "" {
+		return errors.New("should not be empty")
+	}
+
+	if len(s) > maxTagValueLength {
+		return errors.New("must be 256 characters or less")
 	}
 
-	return option == optYes, nil
+	return nil
+}
+
+func ValidateGitURL(s string) error {
+	_, _, err := gitutil.ParseGitRepoURL(s)
+
+	return err
+}
+
+func ValidateGenericInputNonEmpty(s string) error {
+	if s == "" {
+		return errors.New("should not be empty")
+	}
+
+	return ValidateGenericInput(s)
+}
+
+func ValidateGenericInput(s string) error {
```

Comment:
```suggestion
func ValidateShortInput(s string) error {
```

Maybe?  Just from the method name I wouldn't be sure why we would force a generic input to 256 characters.

Reply:
done